azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Video Analyzer %q was not found in Resource Group %q - removing from state", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Video Analyzer %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	props := resp.PropertiesType
+	if props != nil {
+		accounts := flattenVideoAnalyzerStorageAccounts(props.StorageAccounts)
+		if err := d.Set("storage_account", accounts); err != nil {
+			return fmt.Errorf("flattening `storage_account`: %s", err)
+		}
+	}
+
+	flattenedIdentity, err := flattenAzureRmVideoServiceIdentity(resp.Identity)
+	if err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
+	}
+
+	if err := d.Set("identity", flattenedIdentity); err != nil {
+		return fmt.Errorf("flattening `identity`: %s", err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceVideoAnalyzerDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VideoAnalyzerID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("issuing AzureRM delete request for Video Analyzer '%s': %+v", id.Name, err)
+	}
+
+	return nil
+}
+
+func expandVideoAnalyzerStorageAccounts(d *pluginsdk.ResourceData) *[]videoanalyzer.StorageAccount {
+	storageAccountRaw := d.Get("storage_account").([]interface{})[0].(map[string]interface{})
+
+	results := []videoanalyzer.StorageAccount{
+		{
+			ID: utils.String(storageAccountRaw["id"].(string)),
+			Identity: &videoanalyzer.ResourceIdentity{
+				UserAssignedIdentity: utils.String(storageAccountRaw["identity_id"].(string)),
+			},
+		},
+	}
+
+	return &results
+}
+
+func flattenVideoAnalyzerStorageAccounts(input *[]videoanalyzer.StorageAccount) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, storageAccount := range *input {
+		output := make(map[string]interface{})
+
+		if storageAccount.ID != nil {
+			output["id"] = *storageAccount.ID
+		}
+
+		if storageAccount.Identity != nil {
+			output["identity_id"] = *storageAccount.Identity.UserAssignedIdentity
+		}
+
+		results = append(results, output)
+	}
+
+	return results
+}
+
+func expandAzureRmVideoAnalyzerIdentity(d *pluginsdk.ResourceData) (*videoanalyzer.Identity, error) {
+	identityRaw := d.Get("identity").([]interface{})
+	if identityRaw[0] == nil {
+		return nil, nil
```

Comment:
since the `identity` block is required this shouldn't be possible, so we could raise an error here I guess?

```suggestion
		return nil, fmt.Errorf("an `identity` block is required")
```